src/rules/function-url-quotes/__tests__/index.js

```diff
@@ -145,6 +145,10 @@ testRule(rule, {
     code: "@font-face { font-family: 'foo'; src: url($variable + \"foo.ttf\" + $variable); }",
   }, {
     code: "a { background: url(\"/images/my_image@2x.png\") }",
+  }, {
+    code: "@-moz-document url-prefix() {}",
+  }, {
+    code: "a { background: url() }",
```

Comment:
Good plan @jeddy3

> On Sep 8, 2016, at 8:35 AM, Richard Hallows <notifications@github.com> wrote:
> 
> In src/rules/function-url-quotes/__tests__/index.js:
> 
> > @@ -145,6 +145,10 @@ testRule(rule, {
> >      code: "@font-face { font-family: 'foo'; src: url($variable + \"foo.ttf\" + $variable); }",
> >    }, {
> >      code: "a { background: url(\"/images/my_image@2x.png\") }",
> > +  }, {
> > +    code: "@-moz-document url-prefix() {}",
> > +  }, {
> > +    code: "a { background: url() }",
> @jeddy3 what do you think?
> 
> If, as @evilebottnawi says, url("") and url-prefix('') are valid CSS then I think the existing behaviour of this rule is correct.
> 
> I think what @Nitive is after is an optional secondary option to make this behaviour more lenient for when there is no argument e.g. "except: ["empty"].
> 
> For example, given: "function-url-quotes": [ "always" ] the following is valid:
> 
> @-moz-document url-prefix("") {
>   a {
>     background: url("x.jpg");
>   }
> }
> Given "function-url-quotes": [ "always" ], { "except": ["empty"] } the following is valid:
> 
> @-moz-document url-prefix() {
>   a {
>     background: url("x.jpg");
>   }
> }
> I also think changing the rule message from "Expected quotes around url argument" to just "Expected quotes" will help clarify the default behaviour of this rule as the "around url argument" bit is funky.
> 
> Maybe to create parameter no-quotes-for-empty?
> 
> They'll be no need for such a rule if we use an except optional secondary option, which reverses the primary option for the specified pattern.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>